plugins/aws_sts/lib/samson_aws_sts/samson_plugin.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module SamsonAwsSts
+  class Engine < Rails::Engine
+  end
+
+  class << self
+    def deploy_env_vars(sts_client:, deploy:)
+      if deploy.stage.aws_sts_iam_role_arn.blank?
+        return {}
+      end
+
+      creds = sts_client.assume_role(
+        role_arn: deploy.stage.aws_sts_iam_role_arn,
+        role_session_name: session_name(deploy)
+      ).credentials
```

Comment:
You might want a mechanism to set `duration_seconds` for this call. Either to allow for shorter-lived credentials for security, or for long-lived credentials if your deploy takes an exceptionally long time.